alws/crud/build_node.py

```diff
@@ -236,31 +238,41 @@ def get_repo(repo_arch, is_debug):
         if built_srpm_url is not None:
             task_query = select(models.BuildTask.build_id).where(
                 models.BuildTask.id == task_id).scalar_subquery()
-            srpm = next(
-                item for item in task_artifacts if item.arch == 'src'
-                and item.type == 'rpm'
-            )
-            srpm_query = select(models.SourceRpm).where(
-                models.SourceRpm.build_id == task_query).options(
-                selectinload(models.SourceRpm.artifact)
-            )
-            res = await db.execute(srpm_query)
-            srpms = res.scalars().all()
-            srpm_href = next(
-                i.artifact.href for i in srpms if i.artifact.name == srpm.name
-            )
-            srpm_info = await pulp_client.get_rpm_package(
-                srpm_href, include_fields=pkg_fields)
+            srpms = [item for item in task_artifacts if item.arch == 'src'
+                     and item.type == 'rpm']
+            if srpms:
```

Comment:
srpm = next(iter(srpms), None)

Reply:
also we can wrap list comprehension/generator with `next` and don't use `iter`